core/task_executor.go

```diff
@@ -0,0 +1,280 @@
+package core
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/anoideaopen/foundation/core/cachestub"
+	"github.com/anoideaopen/foundation/core/reflectx"
+	"github.com/anoideaopen/foundation/core/telemetry"
+	"github.com/anoideaopen/foundation/core/types"
+	"github.com/anoideaopen/foundation/proto"
+	pb "github.com/golang/protobuf/proto" //nolint:staticcheck
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const ExecuteTasksEvent = "executeTasks"
+
+var ErrTasksNotFound = errors.New("no tasks found")
+
+// ExecuteTaskRequest represents a request to execute a group of tasks.
+type ExecuteTaskRequest struct {
```

Comment:
Tasks vs Task

Reply:
I fixed, you are right, this is tasks ExecuteTasksRequest like method name

cons ExecuteTasks         = "executeTasks"